### PR TITLE
Add --dry-run Option for tkn task start

### LINKED ISSUE #654: Add --dry-run Option for tkn task start
Similar to #372 but for TaskRuns

# Expected Behavior

#643 is adding the ability to view a PipelineRun without creating the PipelineRun via a `--dry-run` option with `tkn pipeline start`. This same behavior should exist for `tkn task start` with viewing a TaskRun. 

This should be able to follow the design in #643 where `tkn task start <taskName> --dry-run`  produces yaml stdout of the TaskRun without actually creating the TaskRun.

There should also be an `--output` option that can be used with `--dry-run` to specify `json` or `yaml` output formats to display the TaskRun.

### PR DESCRIPTION
**Pull request by danielhelfand**
Closes #654 

Similar to #643, this pull request adds the ability to do a dry run of a taskrun with `tkn task start`. A notable issue with this, is that steps won't be included with this approach since it mainly relies on a `TaskRef` to start tasks, but it does show information on inputs, outputs, and params as well as the task reference being used. Using the `-f` option does provide information about steps since it does not require a TaskRef.

This pull request also makes a small change for the `--timeout` option, which is currently not available with `-f` due to only defining it when a task name argument is provided. 

The last change in this pr is getting rid of the `-o` shorthand for output with `--dry-run` for `tkn pipeline start --dry-run`. `-o` is not available with `tkn task start` because of outputs so I think it makes sense to keep it consistent across all commands. Only specifying `--output` works now. 

# Submitter Checklist

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Regenerate the manpages and docs with `make docs` and `make man` if needed.
- [x] Run the code checkers with `make check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

# Release Notes

```
Add --dry-run option for tkn task start
```

### CODE REVIEW

File: pkg/cmd/task/start.go [hunk 1]
```diff
@@ -186,6 +194,7 @@ func startTask(opt startOptions, args []string) error {
 			TaskSpec: &task.Spec,
 		}
 	}
+	tr.Spec.Timeout = &metav1.Duration{Duration: timeoutSeconds}
```
Comment (danielhelfand) [inline pkg/cmd/task/start.go]: Moving setting `Timeout` here so that `-f` can set timeout for TaskRun.

File: pkg/cmd/pipeline/start.go [hunk 2]
```diff
@@ -153,7 +153,7 @@ like cat,foo,bar
 	c.Flags().BoolVarP(&opt.Last, "last", "L", false, "re-run the pipeline using last pipelinerun values")
 	c.Flags().StringSliceVarP(&opt.Labels, "labels", "l", []string{}, "pass labels as label=value.")
 	c.Flags().BoolVarP(&opt.DryRun, "dry-run", "", false, "preview pipelinerun without running it")
-	c.Flags().StringVarP(&opt.Output, "output", "o", "", "format of pipelinerun dry-run (yaml or json)")
+	c.Flags().StringVarP(&opt.Output, "output", "", "", "format of pipelinerun dry-run (yaml or json)")
```
Comment (danielhelfand) [inline pkg/cmd/pipeline/start.go]: Removing to keep consistency across `--dry-run` commands.

File: pkg/cmd/task/start_test.go [hunk 3]
```diff
@@ -898,3 +900,130 @@ func Test_parseRes(t *testing.T) {
 		})
 	}
 }
+
+func TestTaskStart_ExecuteCommand(t *testing.T) {
```
Comment (danielhelfand) [inline pkg/cmd/task/start_test.go]: We should eventually move as many tests in this block as can be moved.

**Comment by chmouel** [comment]
/lgtm
/approve

Thanks 👍🏼🔥

**Pull request merged**